autoload/clap/provider/filer.vim

```diff
@@ -36,13 +29,8 @@ function! s:handle_round_message(message) abort
 
   elseif has_key(decoded, 'result')
     let result = decoded.result
-    if result.total == 0
-      let s:filer_empty_cache[result.dir] = s:DIRECTORY_IS_EMPTY
-      call g:clap.display.set_lines([s:DIRECTORY_IS_EMPTY])
-    else
-      let s:filer_cache[result.dir] = result.entries
-      call g:clap.display.set_lines(result.entries)
-    endif
+    let s:filer_cache[result.dir] = result.entries
+    call g:clap.display.set_lines(result.entries)
```

Comment:
You should inject the `query` candidate here, and `clap#filter#on_typed()` don't have to be changed.